### DISCUSSION TITLE
Probably an impossible request modulation section

### DISCUSSION THREAD

**Discussion by ptowns** [bangcorrupt/hacktribe#39] [Ideas]
So to me one of the biggest missed opportunity of the electribe 2 is putting the modulation "post sequencer" 
So what I wish korg had done was simply allowed the MODULATION section to output midi. Then you could motion record the modulation and move to the next modulation.
So I have worked around this with pure data or axoloti to send lfo to midi cc and then record it in on the electribe. Also would be nice to have velocity as a modulation source. Ie velocity to filter cutoff. Once again I work around this with pure data.

Also huge would be motion record midi cc out!!
If possible. 🤞🏽

**Comment by bangcorrupt** [comment]
Velocity as a motion sequence I think I can do at some point.  It should just need pointing the motion sequence function at the velocity data (maybe).  Using the groove function instead could allow setting the depth of velocity modulation.  

I had a motion sequence outputting as MIDI CC at one point but it needs a lot more work before it's useable (if it ever is).  Motion sequences are interpolated 96 ticks per step, which caused a crash.  Jamming assembly instructions into time critical code is asking for trouble.  

Sending LFO as MIDI would face the same issues.  I'm not sure recording them to the sequencer is worth trying.  Motion sequences record only one value per step, so only slow modulations can be recorded with any accuracy.  It may be worth it for the event recorder where everything is timed to the ms, I haven't looked at how it works yet.  

I will do some work on the modulation section.  There will be a sine wave LFO and I'll try to make things like key sync configurable via SysEx.

**Reply by ptowns** [reply]
Wow, so the modulation outputs 96 times per step! That explains the ability to FM the pitch. So key sync would be for doing fm or just restarting the lfo wave form? 
I remember on the old electribes you could have the motion record set to smooth or trig hold were it would be smooth changes or jump. Guessing the E2 is locked to smooth?

**Reply by bangcorrupt** [reply]
I'm not sure about the resolution of the modulation, motion sequences are interpolated at 96 ticks per step.  

Key sync retriggers the modulation with each note-on.  

Motion sequences can be set to trigger hold on the e2.  I don't want to try adding MIDI motion sequences until I know it will work with interpolated sequences.  I think lowering the interpolation to 6 ticks per step (24ppqn) and maybe making it dependent on the BPM could help.
